src/Results/DTO/GenerativeAiResult.php

```diff
@@ -0,0 +1,313 @@
+     * Constructor.
+     *
+     * @since n.e.x.t
+     *
+     * @param string $id Unique identifier for this result.
+     * @param Candidate[] $candidates The generated candidates.
+     * @param TokenUsage $tokenUsage Token usage statistics.
+     * @param array<string, mixed> $providerMetadata Provider-specific metadata.
+     */
+    public function __construct(string $id, array $candidates, TokenUsage $tokenUsage, array $providerMetadata = [])
+    {
+        $this->id = $id;
+        $this->candidates = $candidates;
+        $this->tokenUsage = $tokenUsage;
+        $this->providerMetadata = $providerMetadata;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @since n.e.x.t
+     */
+    public function getId(): string
+    {
+        return $this->id;
+    }
+
+    /**
+     * Gets the generated candidates.
+     *
+     * @since n.e.x.t
+     *
+     * @return Candidate[] The candidates.
+     */
+    public function getCandidates(): array
+    {
+        return $this->candidates;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @since n.e.x.t
+     */
+    public function getTokenUsage(): TokenUsage
+    {
+        return $this->tokenUsage;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @since n.e.x.t
+     */
+    public function getProviderMetadata(): array
+    {
+        return $this->providerMetadata;
+    }
+
+    /**
+     * Converts the first candidate to text.
+     *
+     * @since n.e.x.t
+     *
+     * @return string The text content.
+     * @throws \RuntimeException If no candidates or no text content.
+     */
+    public function toText(): string
+    {
+        if (empty($this->candidates)) {
+            throw new \RuntimeException('No candidates available');
+        }
+
+        $message = $this->candidates[0]->getMessage();
+        foreach ($message->getParts() as $part) {
+            if ($part->getType()->equals(MessagePartTypeEnum::text()) && $part->getText() !== null) {
+                return $part->getText();
+            }
+        }
+
+        throw new \RuntimeException('No text content found in first candidate');
+    }
+
+    /**
+     * Converts the first candidate to an image file.
+     *
+     * @since n.e.x.t
+     *
+     * @return FileInterface The image file.
+     * @throws \RuntimeException If no candidates or no image content.
+     */
+    public function toImageFile(): FileInterface
+    {
+        if (empty($this->candidates)) {
+            throw new \RuntimeException('No candidates available');
+        }
+
+        $message = $this->candidates[0]->getMessage();
+        foreach ($message->getParts() as $part) {
+            if ($part->getType()->equals(MessagePartTypeEnum::inlineFile()) && $part->getInlineFile() !== null) {
+                return $part->getInlineFile();
+            }
+            if ($part->getType()->equals(MessagePartTypeEnum::remoteFile()) && $part->getRemoteFile() !== null) {
+                return $part->getRemoteFile();
+            }
```

Comment:
These should also check for the MIME type of the respective file to ensure it's an image?

I mentioned in another comment adding a `MimeTypeUtil` class, I think that class could also be used here, e.g. with a method `isImage(FileInterface $file): bool`.